frontend/packages/aquascope-editor/src/lib.ts

```diff
@@ -212,15 +174,31 @@ export class Editor {
     readonly serverUrl: URL = DEFAULT_SERVER_URL,
     readonly noInteract: boolean = false
   ) {
-    let parseResult = parseWithDelimiters(initialCode, [["`[", "]`"]]);
-    if (parseResult.type == "err") throw new Error(parseResult.error);
-
-    let resetMarkedRangesOnEdit = EditorView.updateListener.of(() => {
-      this.markedRanges = [];
-    });
+    let interpParseResult = parseWithDelimiters(initialCode, [["`[", "]`"]]);
+    if (interpParseResult.type == "err")
+      throw new Error(interpParseResult.error);
+
+    this.focusStepConfigs = {};
+    let stepParseResult = parseWithDelimiters(
```

Comment:
Is there a reason we're parsing a second time? The `parseWithDelimiters` function was designed such that you can pass it multiple sets of delimiters at once.